crowbar_framework/app/controllers/repositories_controller.rb

```diff
@@ -0,0 +1,38 @@
+#
+# Copyright 2015, SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+class RepositoriesController < ApplicationController
+  before_filter :repository_registry
```

Comment:
Yes, I don't like this - it hard-couples the controller with an implementation detail in the `Repository` class.  The JIT-based implicit loading you had before was better.  My only quibble was with the method name, which I think should be `#registry` for reasons already explained.